classes/experiment_manager.php

```diff
@@ -112,11 +117,59 @@ public function delete_condition($eid, $condset) {
             $sqlcondition = $DB->sql_compare_text($condset, strlen($condset));
             $DB->execute('DELETE FROM {tool_abconfig_condition} WHERE experiment = ? AND condset = ?', array($eid, $sqlcondition));
         }
+        self::invalidate_experiment_cache();
     }
 
     public function delete_all_conditions($eid) {
         global $DB;
         $DB->delete_records('tool_abconfig_condition', array('experiment' => $eid));
+        self::invalidate_experiment_cache();
+    }
+
+    // ===============================================CACHING FUNCTIONS======================================
+    private function invalidate_experiment_cache() {
+        \cache_helper::invalidate_by_definition('tool_abconfig', 'experiments', array(), array('allexperiment'));
+    }
+
+    public function get_experiments() {
+        $cache = cache::make('tool_abconfig', 'experiments');
+        return $cache->get('allexperiment');
+    }
+
+    public function get_active_request() {
+        $experiments = self::get_experiments();
+
+        $return = array();
+        foreach ($experiments as $experiment) {
```

Comment:
these are fine as is, but a neat function for stuff like this is

https://www.php.net/manual/en/function.array-filter.php